Assets/Scripts/KatamariMovement.cs

```diff
@@ -367,10 +375,14 @@ public void UpdateTimeStateMachine()
                     bool isHitSuccessful = resourceManager.tryToHit();
                     if (isHitSuccessful)
                     {
-                        movementState = StateMachine.normalSpeed;
-
                         rb.constraints = RigidbodyConstraints.None;
-                        forceMode = ForceMode.Force;
+
+                        // Make sure force is calculated in golf state
+                        ForceMode forceMode = ForceMode.Impulse;
+                        rb.AddForce(CalculateHitVector(1), forceMode);
+                        rb.AddTorque(CalculateRollVector(1), forceMode);
```

Comment:
We really should apply the impulse force here exactly once during the state transition instead of it happening in another asynchronous loop in `FixedUpdate`.

Reply:
agree